lib/matplotlib/_constrained_layout.py

```diff
@@ -245,6 +254,37 @@ def check_no_collapsed_axes(layoutgrids, fig):
     return True
 
 
+def compress_fixed_aspect(layoutgrids, fig):
+    extraw = dict()
+    extrah = dict()
+    for ax in fig.axes:
+        if hasattr(ax, 'get_subplotspec'):
+            actual = ax.get_position(original=False)
+            ax.apply_aspect()
+            sub = ax.get_subplotspec()
+            gs = sub.get_gridspec()
+            if gs not in extraw.keys():
+                extraw[gs] = np.zeros(gs.ncols)
+                extrah[gs] = np.zeros(gs.nrows)
+            orig = ax.get_position(original=True)
+            actual = ax.get_position(original=False)
+            dw = orig.width - actual.width
+            if dw > 0:
+                for i in sub.colspan:
+                    extraw[gs][i] = max(extraw[gs][i], dw)
```

Comment:
I think you can do this with NumPy?
```suggestion
                extraw[gs][sub.colspan] = np.maximum(extraw[gs][sub.colspan], dw)
```

Reply:
That seems to work.  OTOH, I'm not sure its that readable?